h/services/jwt.py

```diff
@@ -1,35 +1,94 @@
-import datetime
+import secrets
+from dataclasses import asdict, dataclass
+from datetime import UTC, datetime, timedelta
+from enum import StrEnum
 from functools import lru_cache
 from typing import Any
 
 import jwt
 from jwt import PyJWKClient
-from jwt.exceptions import PyJWTError
+from jwt.exceptions import InvalidTokenError, PyJWTError
 
 JWK_CLIENT_TIMEOUT = 10
 
 
-class TokenValidationError(Exception):
+class JWTDecodeError(Exception):
```

Comment:
Renamed this exception class: it covers all kinds of JWT decoding errors (not just validation errors) and it should have "JWT" in the name of make it clear what it is.